pkg/relayer/session/session_test.go

```diff
@@ -1,3 +1,108 @@
 package session_test
 
-// TODO: Add tests to the relayerSessionsManager logic
+import (
+	"context"
+	"testing"
+	"time"
+
+	"cosmossdk.io/depinject"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/observable/channel"
+	"github.com/pokt-network/poktroll/pkg/relayer"
+	"github.com/pokt-network/poktroll/pkg/relayer/miner"
+	"github.com/pokt-network/poktroll/pkg/relayer/session"
+	"github.com/pokt-network/poktroll/testutil/testclient/testblock"
+	"github.com/pokt-network/poktroll/testutil/testclient/testsupplier"
+	"github.com/pokt-network/poktroll/testutil/testrelayer"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+)
+
+func TestRelayerSessionsManager_Start(t *testing.T) {
+	const (
+		sessionStartHeight = 1
+		sessionEndHeight   = 2
+	)
+	var (
+		zeroByteSlice = []byte{0}
+		ctx           = context.Background()
+	)
+
+	// Set up dependencies.
+	blocksObs, blockPublishCh := channel.NewReplayObservable[client.Block](ctx, 1)
+	blockClient := testblock.NewAnyTimesCommittedBlocksSequenceBlockClient(t, blocksObs)
+	supplierClient := testsupplier.NewOneTimeClaimProofSupplierClient(ctx, t)
+
+	deps := depinject.Supply(blockClient, supplierClient)
+	storesDirectoryOpt := testrelayer.WithTempStoresDirectory(t)
+
+	// Create a new relayer sessions manager.
+	relayerSessionsManager, err := session.NewRelayerSessions(ctx, deps, storesDirectoryOpt)
+	require.NoError(t, err)
+	require.NotNil(t, relayerSessionsManager)
+
+	// Pass a mined relays observable via InsertRelays.
+	mrObs, minedRelaysPublishCh := channel.NewObservable[*relayer.MinedRelay]()
+	minedRelaysObs := relayer.MinedRelaysObservable(mrObs)
```

Comment:
Initial experimentation did not yield great results. The repo is a fork of what we've been using thus far so I figured it would either just work or not, and it seems like the latter is the case. I'm confident that my experimentation was done with the uber `mockgen` binary because `mockgen -version` returned  `v0.3.0`, which is the current latest version.

Buffering these screenshots here for a potential issue on github.com/uber-go/mock:

![image](https://github.com/pokt-network/poktroll/assets/600733/8c0ba2ac-13e1-4078-a0c7-933e5734b073)
> **Fig. 1**: an interface with a method which includes a generic type from another package.

![image](https://github.com/pokt-network/poktroll/assets/600733/a5975735-3116-412f-85bf-b8407982d647)
> **Fig. 2**: invalid go output produced by ` mockgen -destination=../../testutil/mockrelayer/relayer_sessions_manager_mock.go -package=mockrelayer . RelayerSessionsManager` (from the `pkg/relayer` directory).

Reply:
@bryanchriswhite Can you make sure to create a ticket to upgrade to uber's mock? I don't think the migration is trivial, but given that the repo we're using is archived and this is the maintained fork (by a large cap), I think it's the way to go.